source/dynamical_systems/test/tests/test_linear.cpp

```diff
@@ -1,51 +1,125 @@
 #include <vector>
 #include <gtest/gtest.h>
 #include "dynamical_systems/Linear.hpp"
+#include "dynamical_systems/exceptions/EmptyBaseFrameException.hpp"
+#include "dynamical_systems/exceptions/EmptyAttractorException.hpp"
 
+#include "state_representation/exceptions/EmptyStateException.hpp"
+#include "state_representation/exceptions/IncompatibleStatesException.hpp"
 #include "state_representation/exceptions/IncompatibleReferenceFramesException.hpp"
 
+using namespace state_representation;
+using namespace dynamical_systems;
 using namespace std::literals::chrono_literals;
 
 class LinearDSTest : public testing::Test {
 protected:
   void SetUp() override {
-    current_pose = state_representation::CartesianPose("A", 10 * Eigen::Vector3d::Random());
-    target_pose = state_representation::CartesianPose("B", 10 * Eigen::Vector3d::Random(),
-                                                     Eigen::Quaterniond::UnitRandom());
+    current_pose = CartesianPose("A", 10 * Eigen::Vector3d::Random());
+    target_pose = CartesianPose("B", 10 * Eigen::Vector3d::Random(), Eigen::Quaterniond::UnitRandom());
   }
   void print_current_and_target_pose() {
     std::cout << current_pose << std::endl;
     std::cout << target_pose << std::endl;
     std::cout << abs(current_pose.get_orientation().dot(target_pose.get_orientation())) << std::endl;
   }
 
-  void test_closeness() {
-    // print_current_and_target_pose();
-    EXPECT_NEAR((current_pose.get_position() - target_pose.get_position()).norm(), 0.0f, linear_tol);
-    EXPECT_NEAR(current_pose.get_orientation().angularDistance(target_pose.get_orientation()), 0.0f, angular_tol);
-  }
-
-  state_representation::CartesianPose current_pose;
-  state_representation::CartesianPose target_pose;
+  CartesianPose current_pose;
+  CartesianPose target_pose;
   unsigned int nb_steps = 100;
   std::chrono::milliseconds dt = 100ms;
   double linear_tol = 1e-3;
   double angular_tol = 1e-3;
 };
 
+TEST_F(LinearDSTest, EmptyConstructorCartesianState) {
+  // construct empty cartesian state DS
+  CartesianPose attractor = CartesianPose::Identity("CAttractor", "A");
+  Linear<CartesianState> ds;
+  // base frame and attractor should be empty
+  EXPECT_TRUE(ds.get_base_frame().is_empty());
+  EXPECT_TRUE(ds.get_attractor().is_empty());
+  ds.set_attractor(attractor);
+  EXPECT_FALSE(ds.get_attractor().is_empty());
+  // when attractor was set without a base frame, expect base frame to be identity with name / reference_frame of attractor
```

Comment:
I guess it's implied by the subsequent expects, but it might be nice to have an `EXPECT_FALSE(get_base_frame().is_empty());` here for explicitness. (same for JointState test)